src/services/publishing/utils/explorer.tsx

```diff
@@ -0,0 +1,197 @@
+import axios, { AxiosResponse } from "axios";
+import { ETHERSCAN_API_KEY } from "../../../config";
+import { NetworkObject } from "../../../types";
+interface ExplorerHeaders {
+  "Content-Type": string;
+}
+
+interface ExploreParams {
+  module: string;
+  action: string;
+  address: string;
+  startblock: string;
+  endblock: string;
+  page: string;
+  offset: string;
+  sort: string;
+  apikey: string;
+}
+
+interface NetworkAPIDetails {
+  apiKey: string;
+  hostname: string;
+}
+
+interface SimplifiedResponse {
+  status: number;
+  reason: string;
+  address: string;
+}
+
+const getHeaders = (): ExplorerHeaders => {
+  const headers = {
+    "Content-Type": "application/json",
+  } as ExplorerHeaders;
+
+  return headers;
+};
+
+const getParams = (contractAddress: string, apiKey: string): ExploreParams => {
+  const headers = {
+    module: "account",
+    action: "txlist",
+    startblock: "0",
+    endblock: "99999999",
+    page: "1",
+    offset: "1",
+    sort: "asc",
+  } as ExploreParams;
+
+  if (apiKey) {
+    headers["apikey"] = apiKey;
+  }
+
+  if (contractAddress) {
+    headers["address"] = contractAddress;
+  }
+
+  return headers;
+};
+
+const getNetworkAPIDetails = (network: NetworkObject): NetworkAPIDetails => {
+  // const chain: string = network.network.chain;
+  const chainId: number = parseInt(network.network.chainId);
+  let url = "";
+  let apiKey = "";
+  let hostnameNetwork = "";
+  let chainType = "";
+
+  if (chainId == 1 || chainId == 3 || chainId == 4 || chainId == 5 || chainId == 42) {
+    hostnameNetwork = "";
+    chainType = "ETH";
+    if (chainId == 3) {
+      hostnameNetwork = "-ropsten";
+    } else if (chainId == 4) {
+      hostnameNetwork = "-rinkeby";
+    } else if (chainId == 5) {
+      hostnameNetwork = "-goerli";
+    } else if (chainId == 42) {
+      hostnameNetwork = "-kovan";
+    }
+    url = `https://api${hostnameNetwork}.etherscan.io`;
+  } else if (chainId == 137 || chainId == 80001) {
+    hostnameNetwork = "";
+    chainType = "MATIC";
+    if (chainId == 80001) {
+      hostnameNetwork = "-testnet";
+    }
+    url = `https://api${hostnameNetwork}.polygonscan.com`;
+  } else {
+    url = `https://api.etherscan.io`;
```

Comment:
etherscan api free and got any limit? may want to shift these endpoint to config variables level